openam-oauth2/src/main/java/org/forgerock/oauth2/core/ResourceOwnerSessionValidator.java

```diff
@@ -48,19 +53,11 @@
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 
-import com.iplanet.sso.SSOException;
-import com.iplanet.sso.SSOToken;
-import com.iplanet.sso.SSOTokenManager;
-import com.sun.identity.authentication.util.ISAuthConstants;
-import com.sun.identity.idm.AMIdentity;
-import com.sun.identity.idm.IdUtils;
-import com.sun.identity.security.AdminTokenAction;
-import com.sun.identity.shared.Constants;
-import com.sun.identity.shared.debug.Debug;
-
-import freemarker.template.Template;
-import freemarker.template.TemplateException;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.servlet.http.HttpServletRequest;
 
+import org.apache.http.client.utils.URIBuilder;
```

Comment:
I am not sure it is acceptable to use Apache HTTPComponents' library in this class. It is not a direct dependency nor a transitive dependency of openam-oauth2 module.

Reply:
Well, I thought it was fine, but now I'm not convinced either. I have refactored the check to use plain Java.